### ISSUE TITLE
Integration with Telescope-orgmode.nvim for fuzzy filtering

### ISSUE THREAD

**Issue by seflue** [chipsenkbeil/org-roam.nvim#65]
Org-roam provides filtering over existing nodes in various contexts. While the built-in filter widget is already quite convenient, because it allows for substring filtering at an arbitrary position, for seasoned telescope users it is missing the feature of fuzzy search and also breaks a bit with the muscle memory when selecting one of the proposed entries.

I am thinking about an integration of org-roam with Telescope-orgmode.nvim, so I would like to discuss, how this could be done with minimal effort in a clean way.

What I learned from the code, that there is a method `roam.ui.select_node`, which provides the built-in widget. It would be great, if we could make this configurable via options. My idea would be to wrap this in a configurable hook function, which by default calls the built-in widget, but which the user also can override via configuration to call a function from Telescope-orgmode.nvim which would have the same signature, but provides all the features we know and like from Telescope.

@chipsenkbeil I would happily implement this by myself, but I would be grateful, if you can share your thoughts on how this extension point for external plugins should look like from your point of view and what the entry points for an implementation would be.

**Comment by chipsenkbeil** [comment]
@seflue yeah, I had given this a little thought in the past, but it was out of scope for what I personally needed, so I didn't implement it myself. Totally happy to support you in doing this yourself. :smile:

Let me get back to you later after I reread parts of the codebase and address some of the other conversations first.

**Comment by bi0ha2ard** [comment]
I built https://github.com/bi0ha2ard/telescope-org_roam.nvim for myself a while back.

It's not exactly what you're talking about because it doesn't replace `roam.ui.select_node`, but it can replace `find_node` and links/backlinks.
